docs: rewrite about-checksums.md to STE rules (#2024)
apache/maven-resolver: src/site/markdown/about-checksums.md

under the License.
-->

Maven Resolver uses checksums to verify the integrity of downloaded artifacts and metadata.
Checksums exist in repositories next to the target file.
The file extension identifies the checksum algorithm that produced the checksum.
Most Maven repositories contain SHA-1 and MD5 checksums by default.
Maven Resolver also produces these checksums by default.
Checksums only provide integrity verification. They do not provide security or trust.
They do not protect against man-in-the-middle or supply chain attacks.

In the past, Maven Resolver used `java.security.MessageDigest` to calculate checksums.
The Java Cryptography Architecture provides secure one-way hashes.

> Maven Resolver used these secure hashes to verify transport integrity.

Reviewer: The original's "(mis)used" was intentional editorial commentary — it acknowledged that applying cryptographic hash functions as transport checksums was a design shortcut, which motivates the SPI section that follows. Dropping it entirely loses that context. Consider preserving the nuance, e.g.:

> Maven Resolver repurposed these secure hashes as checksums for transport integrity validation.

Secure hashes work as checksums, but cryptographically safe algorithms
require many more CPU cycles to calculate than a simple checksum.

Some users state that specific algorithms are unsafe or deprecated.
This argument does not apply to Maven Resolver because checksums do not provide security.
This fact is true for the SHA-1 algorithm and the MD5 algorithm.
Industry still uses both algorithms today to verify transport integrity and to detect errors.

> To prove that artifacts have not been tampered with, you need signatures such as

Reviewer: Minor: "prove" is stronger than the original "trust." GPG signatures provide cryptographic assurance from a trusted signer — they establish trust, not mathematical proof. The original word was more precise.

```suggestion
To trust that artifacts have not been tampered with, you need signatures such as
```

those provided by the 
[Maven GPG Plugin](https://maven.apache.org/plugins/maven-gpg-plugin/).

## Checksum Algorithms SPI

> The user configuration enables POMs to specify arbitrary checksum algorithms,

Reviewer: The original was deliberately generic ("one can set them via configuration"). The rewrite introduces "POMs" as the mechanism, but the project's own [expected-checksums.md](https://maven.apache.org/resolver/expected-checksums.html) docs show checksum algorithms are configured via system properties (`aether.layout.maven2.checksumAlgorithms`) and `-D` flags, not POMs.

```suggestion
Configuration allows users to specify arbitrary checksum algorithms,
even if they are not part of the standard Maven process.
```

even if they are not part of the standard Maven process.
Users can also register an alternate provider for Java Cryptography that
supplies a broader set of message digests for checksums.
The Maven Resolver team discourages this.

To control the supported set of checksums, the Maven Resolver team introduced an SPI for checksums.
We no longer expose `MessageDigest` algorithms directly.

> Instead it supports four checksum algorithms:

Reviewer: Minor grammar: missing comma after "Instead" (conjunctive adverb), and the antecedent of "it" is unclear since the preceding sentence's subject is "We."

```suggestion
Instead, the SPI supports four checksum algorithms:
```
